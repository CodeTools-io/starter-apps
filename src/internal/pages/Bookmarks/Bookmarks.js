import React from 'react';
import { Box, Card, Heading } from 'grommet';

export default function Bookmarks({ children, ...props }) {
  return (
    <Box className="Bookmarks" fill {...props}>
      <Card pad="large" background="white" fill>
        <Heading
          level={1}
          margin={{ horizontal: 'auto' }}
          textAlign="center"
          fill
        >
          Coming Soon!
        </Heading>
      </Card>
    </Box>
  );
}
