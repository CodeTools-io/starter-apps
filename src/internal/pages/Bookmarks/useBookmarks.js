import { useMemo } from 'react';
import { orderBy } from 'lodash';

export default function useBookmarks({
  components = [],
  categories = [],
  bookmarks = [],
}) {
  const filteredComponents = useMemo(() => {
    const bookmarked = bookmarks?.reduce((accum, bookmark) => {
      const key = `${bookmark?.categoryId}-${bookmark?.moduleId || 'none'}-${
        bookmark?.componentId
      }`;
      return {
        ...accum,
        [key]: true,
      };
    }, {});

    const filtered = components?.filter((component) => {
      const key = `${component?.categoryId}-${component?.moduleId || 'none'}-${
        component?.id
      }`;

      return bookmarked[key];
    });

    return orderBy(filtered, ['id'], ['asc']);
  }, [bookmarks, components]);

  return useMemo(() => {
    return {
      filteredComponents,
    };
  }, [filteredComponents]);
}
