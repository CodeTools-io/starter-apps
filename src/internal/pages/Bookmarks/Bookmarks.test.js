import React from 'react'
import Bookmarks from './Bookmarks'

describe('Bookmarks', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <Bookmarks>
        <p>Content for the Bookmarks</p>
      </Bookmarks>
    )

    expect(container).toMatchSnapshot()
  })
})

