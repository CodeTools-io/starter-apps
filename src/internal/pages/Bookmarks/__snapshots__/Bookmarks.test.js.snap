// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmarks can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 cxNasQ Bookmarks"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jDoFwS"
    >
      <h1
        class="StyledHeading-sc-1rdh4aw-0 xhldK"
      >
        Coming Soon!
      </h1>
    </div>
  </div>
</div>
`;
